### PR TITLE
Add RFC8081 Media Types

### PR DESCRIPTION
**Pull request by dali99**
[RFC8081](https://tools.ietf.org/html/rfc8081) adds a top level Media Type for fonts, it also adds ttf, otf,woff, and woff2 subtypes along with the more general sfnt and collection

This pull request adds the rfc8081 media types, and removes the old unoffical x-truetype-font mime type

I also changed the truetype mime detection special case to font/ttf instead of x-truetype-font, when seeing both the old and the "new old" mime types. I'm honestly not sure what that function is doing or why truetype fonts in particular have this special case. So feel free to suggest edits to that commit

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -11,7 +11,8 @@
   with `segment_duration == 0` when there's more than one edit list entry. In
   that case mkvmerge was reading the whole content more than once. Fixes
   #2152.
-
```
Comment (mbunkus) [inline NEWS.md]: The two empty lines here are intentional. Please keep them.

**Comment by mbunkus** [comment]
Thanks for the pull request, but I won't merge it as-is. Adding new MIME types for fonts is OK, but using them by default isn't for the aforementioned reasons.

**Comment by dali99** [comment]
I figured implementing #2154 would be good enough for those situations.

But then I'll revert the weird mime type detection thing right?
Is duplicate extensions in the list okay or should they be moved to the old x-truetype-font?

**Comment by dali99** [comment]
Is this acceptable?

**Comment by mbunkus** [comment]
Extensions should only be present for one MIME type. All auto-detection should keep using the old MIME types. Therefore please do the following:

* remove the extensions from the "Add RFC-8081 Media Types" commit
* squash "Add RFC-8081 Media Types" with "Add Credits to top comments"
* squash all commits that modify `NEWS.md`
* drop "Remove duplicate filetype extension", "Change mime type truetype detection to use font/ttf", "Remove old truetype mime type"
* force-push the changed commits

Basically this changes needs just one or maybe two commits (one for adding the new MIME types & one for the NEWS.md addition — but even those two could be a single commit).

> I figured implementing #2154 would be good enough for those situations.

I'd like the default to be compatible with most software out there. You've already experienced that existing software doesn't support `font/…` yourself with mpv. Forcing a lot of users to always set the MIME type manually would be quite annoying.

**Comment by dali99** [comment]
Done, not using official mime types makes questions like, should application/x-truetype-font be used for Open Type Format fonts without truetype glyphs? Should they use application/vnd.ms-opentype? Maybe even application/font-sfnt like MKVToolnix seems to default to at the moment, as noted in mpv-player/mpv#4992

I'm not sure exacly how mkvtoolnix chooses headers, since files with .ttf and .otf get different mimes even though they're in "application/x-truetype-font" in this file.

I kept the new woff extentions since they weren't in there before.

**Comment by mbunkus** [comment]
Thanks for implementing all the changes!

> I'm not sure exacly how mkvtoolnix chooses headers, since files with .ttf and .otf get different mimes even though they're in "application/x-truetype-font" in this file.

mkvmerge uses libmagic for file type detection if it is compiled with it. Only if it's compiled without libmagic or if libmagic detection fails does the built-in MIME type list come into play.

The drop-down boxes are always populated from the built-in list, though.

> I kept the new woff extentions since they weren't in there before.

That's quite alright.

**Comment by dali99** [comment]
Thanks for merging. And thanks a lot for the great software!
